spring-web/src/main/java/org/springframework/web/filter/ForwardedHeaderFilter.java

```diff
@@ -242,6 +251,16 @@ public String getHeader(String name) {
 			this.host = uriComponents.getHost();
 			this.port = (port == -1 ? (this.secure ? 443 : 80) : port);
 
+			UriComponents remoteUriComponents = UriComponentsBuilder.newInstance()
+					.host(request.getRemoteHost())
+					.port(request.getRemotePort())
+					.adaptFromForwardedForHeader(httpRequest.getHeaders())
+					.build();
```

Comment:
It is worth applying some sort of check if this even needs to be done, e.g. "X-Forwarded-For" and/or an indexOf scan to see if "Forward" has a "for" attribute. In hot paths especially we need to reduce Object creation. The same applies for the use of Optional.